Source/Initialization/WarpXAMReXInit.cpp

```diff
@@ -36,11 +36,24 @@ namespace {
             amrex::ParmParse pp_amr("amr");
             pp_amr.add("blocking_factor", 1);
         }
+
+        // Here we override the default tiling option for particles, which is always
+        // "false" in AMReX, to "false" if compiling for GPU execution and "true"
+        // if compiling for CPU.
+        {
+            ParmParse pp_particles("particles");
+#ifdef AMREX_USE_GPU
+            bool do_tiling = false; // By default, tiling is off on GPU
+#else
+            bool do_tiling = true;
+#endif
+            pp_particles.add("do_tiling", do_tiling);
```

Comment:
Can we make it as follows so that the user still could override it in inputs and command line argument?
```
pp_particles.queryAdd("do_tiling", do_tiling);
```

I think we should also make a similar change to
```
bool abort_on_out_of_gpu_memory = true; // AMReX' default: false
pp_amrex.query("abort_on_out_of_gpu_memory", abort_on_out_of_gpu_memory);
pp_amrex.add("abort_on_out_of_gpu_memory", abort_on_out_of_gpu_memory);
```
in this function.

Reply:
Yep, the old "query-then-add" logic did also keep the possibility to overwrite with user options :+1: